packages/core/src/controllers/relayer.ts

```diff
@@ -304,6 +335,41 @@ export class Relayer extends IRelayer {
   }
 
   // ---------- Private ----------------------------------------------- //
+  /*
+   * In Node, we must detect when the connection is stalled and terminate it.
+   * The logic is, if we don't receive ping from the relay within a certain time, we terminate the connection.
+   * The timer is refreshed on every message received from the relay.
+   *
+   * In the browser, ping/pong events are not exposed, so the above behaviour is handled by `subscribeToNetworkChange` and `isOnline` functions.
+   */
+  private startPingTimeout() {
+    if (!isNode()) return;
+    try {
+      //@ts-ignore
+      if (this.provider?.connection?.socket) {
+        //@ts-ignore
+        this.provider?.connection?.socket?.once("ping", () => {
+          this.heartbeat();
+        });
+      }
+      this.heartbeat();
+    } catch (e) {
+      this.logger.warn(e);
+    }
+  }
+
+  private heartbeat = () => {
```

Comment:
Feels like we're overloading the term `heartbeat` bc we already have other use cases with this name in our core, would rather it just says what it does:

```suggestion
  private resetPingTimeout = () => {
```